lib/Sema/TypeChecker.h

```diff
@@ -714,12 +714,12 @@ class TypeChecker final : public LazyResolver {
   ///
   /// \returns A BoundGenericType bound to the given arguments, or null on
   /// error.
-  Type applyGenericArguments(Type type,
-                             SourceLoc loc,
-                             DeclContext *dc,
-                             MutableArrayRef<TypeLoc> genericArgs,
-                             bool isGenericSignature,
-                             GenericTypeResolver *resolver);
+  Type applyUnboundGenericArguments(UnboundGenericType *unbound,
```

Comment:
Is it ok to pass an `UnboundGenericType` or should it be wrapped inside a `Type` value?